### PR TITLE
use ThreadPoolExecutor to wait for asyncronous processes as they complete

### PR DESCRIPTION
**Pull request by anthrotype**
And also allow to set max number of worker subprocesses with -j option

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -28,6 +30,22 @@
 
 
 def main():
+    parser = argparse.ArgumentParser()
```
Comment (rsheeter) [inline build.py:33]: We use absl flags everywhere else so I might suggest here too?

File: build.py [hunk 2]
```diff
@@ -28,6 +30,22 @@
...
+    parser.add_argument(
```
Comment (rsheeter) [inline build.py:34]: feels very natural to just pass them in argv, does having them flagged help? (no real complaint, just curious)

Comment (anthrotype) [reply build.py:34]: well, the `configs` argument is "positional", only the -j is optional (I guess that's what you mean by "flagged")

File: build.py [hunk 3]
```diff
@@ -61,41 +79,46 @@ def main():
...
+                    len(font_files) == 1
+                ), f"Expected exactly 1 font file in {working_build_dir}, found {font_files}"
+                src, dst = font_files[0], font_dir / font_files[0].name
+                shutil.copy(src, dst)
+                print(src, "=>", dst)
+            else:
+                print(f"Failed: {cmd_str}")
+                print("See {working_dir.relative_to(self_dir)} for logs")
+
+            before_rmtree = time.monotonic()
```
[the remark this answers is not in the document]
Comment (anthrotype) [reply build.py:120]: it was already there, I'll remove it

**Pull request merged**